### DISCUSSION TITLE
Starting with the latest version, is it impossible to install on a server without Docker?

### DISCUSSION THREAD

**Discussion by kdh1122** [milvus-io/milvus#33496] [Q&A and General discussion] [answered]
I saw on the milvus that it has been updated to support only Docker and Kubernetes. 
Do you plan on not supporting installation on servers in the future?

**Comment by yhmo** [comment, marked as answer]
Milvus-Lite is a lightweight version of Milvus, a no-server tool running locally.
https://milvus.io/docs/milvus_lite.md
https://github.com/milvus-io/milvus-lite
https://milvus.io/blog/introducing-milvus-lite.md

No need to deploy, no need to use docker or k8s. It stores data in a local SQLite. The same client API as Milvus.

**Reply by kdh1122** [reply]
Thank you for answer. So is only Milvus-Lite available?

**Reply by yhmo** [reply]
If you want to run a milvus server without docker, you need to compile it from source code.

Steps to compile and run Milvus locally:

1. Enter the project root path, run this script to set environment paths.
`source ./scripts/setenv.sh`

2. Under the project root path, run this script to install the dependent libraries.
`./scripts/install_deps.sh`


3. Under the project root path, run "make" command to compile milvus.
`make`
Once it successfully compiles, you will see ./bin/milvus under the project root path.


4. Use a text tool to open the ./config/milvus.yaml, replace all the "/var/lib/milvus" to be "/tmp/milvus".
This step is to avoid the permission deny problem during milvus runtime.


5. Start the third-party services by docker, including etcd/minio.
`cd deployments/docker/dev
docker-compose up -d`

6. Back to the project root path, run this script to start a local milvus standalone.
`./scripts/start_standalone.sh`
Use command `ps -e | grep milvus` to check, you will see a milvus process running.

The runtime log will be gernerated under /tmp/standalone.log. If the server successfully starts, you will see log like this:
`---Milvus Proxy successfully initialized and ready to serve!---
[2024/03/21 18:38:57.874 +08:00] [INFO] [components/proxy.go:58] ["Proxy successfully started"]`

**Reply by kdh1122** [reply]
Will I still be able to use this method in future updated versions? Our company is considering whether to use milvus, but since we do not use Docker, we are curious about the future direction of milvus.

**Reply by yhmo** [reply]
Definitely, we will keep pushing Milvus-Lite, as a light-weight vector database to build LLM apps. As mentioned in this blog: https://milvus.io/blog/introducing-milvus-lite.md

**Comment by xiaofan-luan** [comment]
@kdh1122 you should be able to find how to deploy distributed or standalone milvus on milvus.io/docs.

We highly recommend you to use k8s for distributed deployment. Yes you can deployed it with ansible or manually, but it required many knowledge and we don't recommend to do so.

**Comment by xiaofan-luan** [comment]
you can setup milvus.yaml by hand, bring up all the node manually.
